autoscheduler/frontend/src/components/LandingPage/SelectTerm/SelectTerm.tsx

```diff
@@ -8,6 +8,23 @@ import { useDispatch } from 'react-redux';
 import setTerm from '../../../redux/actions/term';
 import * as styles from './SelectTerm.css';
 
+/**
+ * Stateful function that caches the json for an api/terms call, so that the API route
+ * only has to be fetched once.
+ */
+const getTermsJson = ((): () => Promise<any> => {
+  let fetched = false;
+  let fetchedData: Promise<any>;
+
+  return async (): Promise<any> => {
+    if (fetched) return fetchedData;
```

Comment:
Could you remove `fetched` altogether and return `fetchedData` if its truthy?

Reply:
Agreed, the `fetched` variable seems kinda redundant

Reply:
Yeah I was planning on implementing this a bit differently at first but you're right, I made the change